### PR TITLE
Shared docker volumes with EFS

### PR DESCRIPTION
**Pull request by ddollar**
This pull request builds on #810 and uses EFS for the backing store for volumes. To use add volumes to the `volumes:` section of your process entry.

```
$ cat docker-compose.yml | grep volumes: -A2
  volumes:
    - /foo
    - /bar:/tmp/bar

$ convox ps
ID            NAME      RELEASE      STARTED        COMMAND
831925acee0e  database  RLMZNZSKUNA  3 minutes ago  /docker-entrypoint.sh postgres
73bf22eed6bc  web       RLMZNZSKUNA  1 minute ago   bin/web
fe66c945567d  web       RLMZNZSKUNA  3 minutes ago  bin/web

$ convox exec 73bf22eed6bc cat /foo/bar
cat: /foo/bar: No such file or directory

$ convox exec fe66c945567d 'echo qux > /foo/bar'

$ convox exec 73bf22eed6bc cat /foo/bar
qux
```

## Release Playbook
- [x] Rebase against master
- [x] Release branch (20160629185452-efs)
- [x] Pass CI (https://circleci.com/gh/convox/rack/880#build-parameters)
- [ ] Code review
- [ ] Merge into master
- [ ] Release master ()
- [ ] Pass CI ()
- [ ] Update staging rack
- [ ] Edit [Rack release record](https://github.com/convox/rack/releases) and/or update [docs](https://github.com/convox/convox.github.io)
- [ ] Publish release
- [ ] Release CLI

### CODE REVIEW

**Comment by ddollar** [comment]
`20160629152014-efs`

**Comment by ddollar** [comment]
Thoughts on the `convox.volumes.mount` label? Should we just turn this on in a blanket way? I'm actually sort of inclined to say let's just go for it.

**Comment by MiguelMoll** [comment]
> Always mount volumes

Would love for that to be the default. Seems the expected behavior.

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;73vs8Kwt&#x2F;306-efs"> efs</a>

**Comment by ddollar** [comment]
We should probably scope `/volumes` by app name and process type.

File: api/models/templates/app.tmpl [hunk 1]
```diff
@@ -633,15 +634,18 @@
     },
     "Volumes": [
       {{ range .MountableVolumes }}
-        "{{ . }}",
+        {{ if eq .Host "/var/run/docker.sock" }}
```
Comment (nzoschke) [inline api/models/templates/app.tmpl]: is this really the only non /volumes/ mount we expect going forward? I can't think of anything else we explicitly support, just asking out loud. And points to some documentation improvements obviously. Can't find anything about volumes right now.

Comment (ddollar) [inline api/models/templates/app.tmpl]: Yes, this is the only special volume we support. Matt is working on the docs for this stuff.

**Comment by nzoschke** [comment]
In testing I encountered one issue. After the first update, 2 out of 3 instances came up without correctly mounting EFS due to DNS problems:

> mount.nfs: Failed to resolve server us-east-1b.fs-f223e6bb.efs.us-east-1.amazonaws.com: Name or service not known

Waiting a few minutes and restarting the servers with `convox instances keyroll` fixed it.

**Pull request merged**